pygments/lexers/matlab.py

```diff
@@ -131,12 +138,26 @@ class MatlabLexer(RegexLexer):
             # function with no args
             (r'(\s*)([a-zA-Z_]\w*)', bygroups(Text, Name.Function), '#pop'),
         ],
+        'string': [
+            (r"[^']*'", String, '#pop'),
+        ],
+        'commandargs': [
+            ("'[^']*'", String),
+            ("[^';\n]+", String),
+            (";?\n?", Punctuation, '#pop'),
+        ]
     }
 
     def analyse_text(text):
-        if re.match(r'^\s*%', text, re.M):  # comment
+        # function declaration.
+        if next(line for line in text.splitlines()
+                if not re.match(r'^\s*%', text)).strip().startswith('function'):
```

Comment:
I think you can just `re.search(r'^function\b')` with `re.MULTILINE`.

Although this is a bit problematic for safe identification, since e.g. JavaScript functions are also defined that way.  (analyse_text can also be used when no filename is known.)

Maybe you can also match the equals sign which I gather is always part of the function header?

Reply:
I don't think the single regex is equivalent, the point is that the *first* line that isn't fully commented should start with `function`.
No, there isn't always an equal sign (if the function doesn't return anything).

Reply:
ah, I missed that, sorry. Still very possible in Javascript though... maybe exclude if `{` comes later?

Reply:
sure, done